packages/store/docs/store-taxonomy.md

```diff
@@ -0,0 +1,67 @@
+* Composite keys are things like copyArrays, copyRecords, copySets, or copyMaps, that compare by structural key equality.
+
+## Unstructured* vs Schematized
+
+* An unstructued store is restricted only by the representation limitations listed here. They keys must be Keys. The values must be Passable. Scalar stores allow only scalar keys. Persistent stores only allow contents that are well defined after vat death.
+* A schematized store is further restricted by a pattern provided when it was created to serve as its schema. A schematized store will only allow contents that match its schema pattern.
+
+## Long* vs Short Expected Lifetime (WeakStores only)
+
+The expected lifetime of a weak store helps us optimize its representation.
+* If we expect a weak store to outlive most of its keys, we may internally use a JavaScript WeakMap.
+* If we expect the keys in a weak store to outlive the store, we may internally use a JavaScript Map, to avoid the costs often associated indexing a key for JavaScript WeakMaps that are no longer alive. Though keep in mind that this representation denies the garbage collector the opportinity to collect those keys while these expected-to-be-short-lived stores are still alive.
+
+## Heap* vs Virtual
+
+* Like JavaScript Maps and Sets, heap stores live in the JavaScript language heap, limited by the size of heap memory and occupying room in vat snapshots.
+* Virtual collections potentially live outside the heap. A virtual store serializes (marshals) its contents to store it, and unserializes (unmarshals) its contents to retrieve it. Virtual objects (a kind of far object) that are accessible *only* from virtual stores can completely disappear from the heap, to be restored if any serialized reference to it is restored. A virtual store is itself a virtual object.
+
+## Ephemeral* vs Persistent (Heap is always Ephemeral)
+
+Stores only exist in one vat at a time, giving us free transactional integrity. All heap stores are ephemeral. Virtual stores can be ephemeral or persistent.
+* Ephemeral stores only exist in the vat that initially created them, dying when that vat dies. The content of virtual ephemeral stores, can contain references to ephemeral non-virtual remotables or promises.
+* A persistent store can be part of a vat's "estate", to be transfered to that vat's heir/successor on the death of its current vat. A persistent store can only contain keys and values that are still meaningful across such traumas, such as
+    * persistent objects (a kind of virtual object)
+    * Remotes, a kind of remotable designating a far object in another vat
+    * Promises, perhaps. These are tempting because they can be restored into a well-defined severed state, the rejected promise.
+
+    A persistent store may not store a reference to a local non-persistent far object. A persistent store is itself a persistent object.
+
+# Iteration Differences
+
+## Rank Order* vs Altered Rank Order Iteration
```

Comment:
The explanation of altered rank order is promising and something we may want to do someday. But not now. We should not lose the text but it doesn't belong in this doc at this time.